cwf/gateway/integ_tests/assertions.go

```diff
@@ -104,12 +105,36 @@ func (tr *TestRunner) DisconnectAndAssertSuccess(imsi string) {
 
 // Query policy usage records from pipelined, and assert that there is no entry
 // for the IMSI. This means no policy flows are installed for the IMSI.
-func (tr *TestRunner) AssertPolicyEnforcementRecordIsNil(imsi string) {
+func (tr *TestRunner) AssertEnforcementRecordIsEmptyForSub(imsi string) {
 	recordsBySubID, err := tr.GetPolicyUsage()
 	assert.NoError(tr.t, err)
 	assert.Empty(tr.t, recordsBySubID[prependIMSIPrefix(imsi)])
 }
 
+func (tr *TestRunner) AssertPolicyEnforcementRecordIsNotNil(recordsBySubID RecordByIMSI, imsi, ruleID string) {
+	record := recordsBySubID[prependIMSIPrefix(imsi)][ruleID]
+	assert.NotNil(tr.t, record, fmt.Sprintf("No enforcement record for %s %s", imsi, ruleID))
+}
+
+func (tr *TestRunner) AssertPolicyEnforcementRecordIsNil(recordsBySubID RecordByIMSI, imsi, ruleID string) {
+	record := recordsBySubID[prependIMSIPrefix(imsi)][ruleID]
+	assert.Nil(tr.t, record, fmt.Sprintf("Unexpected enforcement record exists for %s %s", imsi, ruleID))
+}
+
+// Assert that the total usage for IMSI, RuleID is between min and max
+func (tr *TestRunner) AssertPolicyEnforcementRecordTotalUsage(imsi, ruleID string, min, max uint64) {
+	recordsBySubID, err := tr.GetPolicyUsage()
+	assert.NoError(tr.t, err)
+	tr.AssertPolicyEnforcementRecordIsNotNil(recordsBySubID, imsi, ruleID)
+	record := recordsBySubID[prependIMSIPrefix(imsi)][ruleID]
+	if record != nil {
+		totalUsage := record.BytesTx + record.BytesRx
+		maxWithErrorMultiplier := uint64(math.Round(float64(max) * float64(BufferMultiplier)))
```

Comment:
I think this is confusing: we are passing a a min and max to function. But inside the function we add even add an extra margin. Shouldn't it be better, either to pass that margin on the function, or just use min and max (no extra buffers)?

This may also help in debugging because you will know exactly how much were your lower and upper limit.

Reply:
Ok, maybe it is more straightforward to allow the caller to adjust for error.